### DISCUSSION TITLE
__turbopack_resolve_absolute_path__ is not a function

### DISCUSSION THREAD

**Discussion by medihack** [vercel/next.js#69834] [Help]
### Summary

When importing [`quickjs-emscripten`](https://github.com/justjake/quickjs-emscripten) I receive the below error when using Turbopack. It works fine when using Next.js 14 without Turbopack. `quickjs-emscripten` uses some WASM stuff under the hood. Could it be that this is the problem and Turbopack doesn't support that? Or can someone give me a hint what else I could try to fix that problem?

The stack trace:
```
Unhandled Runtime Error
TypeError: __turbopack_resolve_absolute_path__ is not a function

Call Stack
get url
file:///workspace/medreporter/.next/static/chunks/node_modules_%40jitl_quickjs-wasmfile-release-sync_dist_cf05b7._.js (15:26)
<unknown>
node_modules/@jitl/quickjs-wasmfile-release-sync/dist/emscripten-module.browser.mjs (5:23)
<unknown>
file:///workspace/medreporter/.next/static/chunks/node_modules_%40jitl_quickjs-wasmfile-release-sync_dist_cf05b7._.js (519:3)
[project]/node_modules/@jitl/quickjs-wasmfile-release-sync/dist/emscripten-module.browser.mjs [client] (ecmascript)
file:///workspace/medreporter/.next/static/chunks/node_modules_%40jitl_quickjs-wasmfile-release-sync_dist_cf05b7._.js (523:3)
<unknown>
file:///workspace/medreporter/.next/static/chunks/%5Bnext%5D_entry_page-loader_ts_c9b2fb._.js (501:27)
runModuleExecutionHooks
file:///workspace/medreporter/.next/static/chunks/%5Bnext%5D_entry_page-loader_ts_c9b2fb._.js (541:9)
instantiateModule
file:///workspace/medreporter/.next/static/chunks/%5Bnext%5D_entry_page-loader_ts_c9b2fb._.js (500:9)
getOrInstantiateModuleFromParent
file:///workspace/medreporter/.next/static/chunks/%5Bnext%5D_entry_page-loader_ts_c9b2fb._.js (573:12)
esmImport
file:///workspace/medreporter/.next/static/chunks/%5Bnext%5D_entry_page-loader_ts_c9b2fb._.js (123:20)
<unknown>
file:///workspace/medreporter/.next/static/chunks/node_modules_261aca._.js (50:16)
async Module.E
file:///workspace/medreporter/.next/static/chunks/node_modules_b246c9._.js (21435:60)
```

The location where this error happens:
```js
const __TURBOPACK__import$2e$meta__ = {
    get url () {
        return `file://${__turbopack_resolve_absolute_path__("node_modules/@jitl/quickjs-wasmfile-release-sync/dist/emscripten-module.browser.mjs")}`;
    }
};
```

**Comment by DuckThom** [comment]
I just got the same error in my Next 14.2.3 project. Although I'm not using QuickJS, perhaps it'll help you or someone else running into the same issue aswell.
Somewhere in our code we were using `import.meta.url` which turbopack (`npm run dev -- --turbo`) does not seem to like. Replacing it with something like `__dirname` seems to fix the error for us.

**Comment by matmill5** [comment]
Having a similar issue when importing onnxruntime-web.

Also, I'd note that the onnx-runtime-web minified bundle in node_modules is using import.meta.url, so may be a source of issue as @DuckThom mentioned.

Also, as @medihack mentioned, onnx-runtime-web is also using WASM technologies.

**Comment by RobertAron** [comment]
Similar problem with `@react-three/uikit`

`"next": "^14.2.4"`

**Reply by SadmanYasar** [reply]
> Similar problem with `@react-three/uikit`
> 
> `"next": "^14.2.4"`

Facing the same issue with uikit and also react three fiber. Have you found any solution?

**Reply by RobertAron** [reply]
Sorry no. I ended up building the component I needed myself.
